content/blog/2017/2017-01-01-conditional.adoc

```diff
@@ -0,0 +1,309 @@
++ echo 'Hello, bitwiseman!'
+Hello, bitwiseman!
+Finished: SUCCESS
+----
+
+This is basic example but the conditional step can contain any regular build step available.
+When combined with other plugins, it can control whether to send notifications,
+gather data from other sources, wait for user feedback, or call other projects.
+
+The Conditional BuildStep plugin does a great job of leveraging strengths of
+the Jenkins UI, and freestyle jobs, and UI-based programming,
+but it is also hampered by their limitations.
+The Jenkins UI can be clunking and confusing at times.
+Like the steps in any freestyle job the logic in these conditional steps is only
+stored and viewable in Jenkins.
+It is not versioned with other product or build code and can't be code reviewed.
+Like any number of UI-based programming it has to make trade-offs between clarity
+and flexibility: more options or clearer presentation.
+There's only so much space on the screen.
+
+== Migrating to Pipeline
+
+// Transitional paragraph to about pipeline
+
+The Pipeline equivalent of the example above is:
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    parameters {
+        choice(
+            // choices must be newline separated
+            choices: ['greeting','silence'].join('\n'),
+            description: '',
+            name: 'REQUESTED_ACTION')
+    }
+
+    stages {
+        stage ('Speak') {
+            when {
+                // Only say hello if a "greeting" is requested
+                expression { return params.REQUESTED_ACTION == 'greeting' }
+            }
+            steps {
+                echo "Hello, bitwiseman!"
+            }
+        }
+    }
+}
+// Script //
+properties ([
+    parameters ([
+        choice (
+            // choices must be newline separated
+            choices: ['greeting','silence'].join('\n'),
+            description: '',
+            name : 'REQUESTED_ACTION')
+    ])
+])
+
+node {
+    stage ('Speak') {
+        // Only say hello if a "greeting" is requested
+        if (params.REQUESTED_ACTION == 'greeting') {
+            echo "Hello, bitwiseman!"
+        }
+    }
+}
+----
+
+Here's the output when I run this project with `REQUESTED_ACTION` set to "greeting":
+
+[source]
+----
+[Pipeline] node
+Running on osx_mbp in /Users/bitwiseman/jenkins/agents/osx_mbp/workspace/pipeline-conditional
+[Pipeline] {
+[Pipeline] stage
+[Pipeline] { (Speak)
+[Pipeline] echo
+Hello, bitwiseman!
+[Pipeline] }
+[Pipeline] // stage
+[Pipeline] }
+[Pipeline] // node
+[Pipeline] End of Pipeline
+Finished: SUCCESS
+----
+
+When at I pass the value "silence", the only change is "Hello, bitwiseman!" is not printed.
+
+The Pipeline code is considerably more compact than the Jenkins UI presentation.
+I had to do a some web searching to find out how to had build
+parameters to Pipeline
+(link:https://issues.jenkins-ci.org/browse/WEBSITE-281[One more task] for our
```

Comment:
See my comment on that ticket,. If the book is out of date, let's just fix that one paragraph in the handbook and merge that instead of including this paragraph in the blog post